e2e/run.js

```diff
@@ -0,0 +1,113 @@
+const { spawn, spawnSync } = require('child_process')
+
+const debug = false
+const createPostgresContainer = true
+const deletePostgresContainer = true
+
+const postgresContainerName = 'indexer-postgres-e2e'
+
+function run(command, args, options) {
+  options = options || {}
+  if (debug) console.log(`${command} \\\n  ${args.join(' \\\n  ')}`)
+  const result = spawnSync(command, args, options)
+  if (result.error) {
+    throw result.error
+  } else if (result.status) {
+    throw new Error(`${command} failed: ${result.status}`)
+  } else if (result.signal) {
+    throw new Error(`${command} exited from signal: ${result.signal}`)
+  }
+  return result
+}
+
+function docker(args, options) {
+  options = options || {}
+  options = {
+    ...options,
+    env: {
+      ...process.env,
+      ...options.env
+    }
+  }
+  return run('docker', args, options)
+}
+
+async function psql(statement, timeoutSeconds) {
+  timeoutSeconds = timeoutSeconds || 0
+  let now = Date.now()
+  const end = now + timeoutSeconds*1000
+
+  while (true) {
+    try {
+      const result = docker(['exec', postgresContainerName,
+                             'psql',
+                             '-t',
+                             '--csv',
+                             '-U', 'postgres',
+                             '-d', 'indexer',
+                             '-c', statement],
+                            {stdio: 'pipe'})
+      if (result.output[1]) {
+        const rows = result.output[1].toString().trim().split('\n')
+        return rows
+      }
+      throw new Error('Empty result')
+    } catch (error) {
+      now = Date.now()
+      if (now > end) {
+        console.log(`Timed out executing '${statement}'`)
+        throw error
+      }
+      await sleep(1000)
+    }
+  }
+}
+
+function sleep(milliseconds) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, milliseconds)
+  })
+}
+
+async function main() {
+  let indexerChildProcess = null
+
+  const exitHandler = () => {
+    if (indexerChildProcess) {
+      indexerChildProcess.kill('SIGINT')
+    }
+    if (deletePostgresContainer) {
+      docker(['rm', '-f', postgresContainerName])
+    }
+  }
+  process.on('exit', exitHandler)
+  process.on('SIGINT', () => process.exit(1))
+  process.on('SIGTERM', () => process.exit(1))  
+  
+  if (createPostgresContainer) {
+    docker(['run',
+            '--name', postgresContainerName,
+            '--rm',
+            '-d',
+            '-p', '5432:5432',
+            '-e', 'POSTGRES_DB=indexer',
+            '-e', 'POSTGRES_PASSWORD=docker',
+            'postgres'])
+  }
+
+  // Ensure DB is up before starting indexer
+  await psql('SELECT 1;', 30)
+  
+  indexerChildProcess = spawn('node', ['./dist/bin/indexer.js'], {stdio: 'inherit'})
+  console.log('Waiting for DB to have some contents...')
+  await psql('SELECT * FROM transfers LIMIT 1;', 30)
+  console.log('DB has some contents')
```

Comment:
I see in the run error logs like `Indexer cUsd_Transfer - Error while handling events Error: CONNECTION ERROR: Couldn't connect to node UNDEFINED.`

However, the test passed 🤔

Reply:
That sounds like a bug somewhere (either in the indexer or in the test code).  I'll take a look.